### PR TITLE
[Tooling] Report an error when no milestone is set to a PR

### PR DESCRIPTION
**Pull request by iangmaia**
This PR should make Danger report an error when no milestone is set to a Pull Request, thus failing the build.

The results should be observed on this very PR.

**Update**: It reports an error and fails the build as expected:
Screenshot 2024-08-27 at 21 12 54

And it switches to a warning when the label `milestone-not-required` is used:
Screenshot 2024-08-27 at 21 15 33

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -54,4 +54,4 @@ labels_checker.check(
 )
 
 # runs the milestone check if this is not a WIP feature and the PR is against the main branch or the release branch
-milestone_checker.check_milestone_due_date(days_before_due: 2) if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature?
+milestone_checker.check_milestone_due_date(days_before_due: 2, report_type_if_no_milestone: :error) if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature? && !github.pr_labels.include?('milestone-exemption')
```
Comment (dangermattic) [inline Dangerfile:57]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-android/blob/2662c486278d2c8fb242a0805e74c0da363a1e1c/Dangerfile#L57"></span>Style/IfUnlessModifier: Modifier form of <code>if</code> makes the line too long.</td>
    </tr>
  </tbody>
</table>

Comment (dangermattic) [inline Dangerfile:57]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-android/blob/2662c486278d2c8fb242a0805e74c0da363a1e1c/Dangerfile#L57"></span>Layout/LineLength: Line is too long. [245/180]</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by malinajirka — APPROVED** [review]
LGTM, thanks!

File: Dangerfile [hunk 2]
```diff
@@ -54,4 +54,10 @@ labels_checker.check(
 )
 
 # runs the milestone check if this is not a WIP feature and the PR is against the main branch or the release branch
-milestone_checker.check_milestone_due_date(days_before_due: 2) if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature?
+if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature?
+  report_type = github.pr_labels.include?('milestone-exemption') ? :warning : :error
```
Comment (mokagio) [inline Dangerfile]: See non-blocking naming suggestion from the iOS counterpart https://github.com/woocommerce/woocommerce-ios/pull/13758#discussion_r1734056934

**Review by mokagio — APPROVED** [review]

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/233">20.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**